### DISCUSSION TITLE
Is it possible to create a library using Mojo and publish it to PyPI for Python users?

### DISCUSSION THREAD

**Discussion by ficapy** [modular/modular#546] [Q&A]
I've read the [Programming Manual](https://docs.modular.com/mojo/programming-manual.html#python-integration), but it only shows examples of Mojo invoking Python libraries. I'm wondering if it's possible to easily package a Mojo-based program and upload it to PyPI for Python users to use. 
It would be great if this information could be added to the Programming Manual.

**Comment by drunkwcodes** [comment]
I'm thinking about it, too.

Calling mojo from python is important.

**Comment by chaochungkuo** [comment]
I also feel that this is an important feature to determine how we want to distribute our work. Can I release my package with some mojo implementation through PyPi? Or, does mojo aims to make another platform for releasing/sharing our work?

**Comment by gamendez98** [comment]
It is not possible at the moment to call mojo from python but the team at modular seem to be working on it

**Reply by msaelices** [reply]
It's on the [roadmap](https://github.com/modularml/mojo/blob/main/docs/roadmap.md). To me, if you want to get massive adoption of Mojo, it would be a priority for Modular.

Imagine the amount of Python projects that could incrementally migrate parts of their code base to optimize the most critical or time-consuming modules.
